src/firmware/app/control/trajectory_planner.h

```diff
@@ -72,232 +71,369 @@ typedef enum TrajectoryPlannerGenerationStatus
+ * Constant arc-length trajectories (not used here) are trajectories where the distance
+ * travelled in each segment in constant.
+ *
+ * @param constant_period_trajectory [out] A constant interpolation period trajectory
+ * generated by the specified path parameters.
+ *
+ * @param path_parameters [in/out] The path parameters that define the trajectory. The
+ * input path_parameters.num_elements defines the number of constant-parameterization
+ * elements, and the returned path_parameters.num_elements defines the number of
+ * constant-period elements in the trajectory,
+ *
+ * @param interpolation_period  [in] The duration of time between successive trajectory
+ * elements. In seconds.
+ *
+ * @return A status indicating whether or not generation was successful
  */
-void static app_trajectory_planner_generatePositionTrajectoryTimeProfile(
-    PositionTrajectoryElement_t* traj_elements, const float num_segments,
-    const float arc_segment_length, float* velocity_profile);
+TrajectoryPlannerGenerationStatus_t
+app_trajectory_planner_generateConstantInterpolationPeriodPositionTrajectory_2(
+    PositionTrajectory_t* constant_period_trajectory,
+    FirmwareRobotPathParameters_t* path_parameters, float interpolation_period);
 
-/***
- *  This function takes in a forwards trajectory annd modifies it in place to become a
- * backwards trajectory NOTE: This function exists to avoid issues outlined in #1322
+/**
+ * Function generates a velocity trajectory composed of X/Y/angular velocities and their
+ * time dependence.
+ *
+ * @param position_trajectory [in] A completely defined position trajectory.
  *
- *  TODO: Remove when #1322 is merged.
+ * @param velocity_trajectory [out] A completely defined velocity trajectory that has
+ * equally time-spaced elements.
  *
- * @param forwards_trajectory This is the trajectory that will be modified in place to
- * become a reverse trajectory
+ * @param num_elements [in] The number of elements in the trajectory.
  */
-void static app_trajectory_planner_reversePositionTrajectoryDirection(
-    PositionTrajectory_t* forwards_trajectory);
+void app_trajectory_planner_generateVelocityTrajectory_2(
+    PositionTrajectory_t* position_trajectory, VelocityTrajectory_t* velocity_trajectory,
+    unsigned int num_elements);
 
 /**
- * This function generates a velocity trajectory that corresponds to the time-optimal
- * velocity to follow a specified path. This profile is based on the input position
- * trajectory
- *
- * @param position_trajectory [in] The position trajectory to build the velocity
- * trajectory from
- * @param velocity_trajectory [out] The velocity trajectory that corresponds to the
- * time-optimal velocity to follow a specified path
+ * Function modifies the value of the argument final speed so that the time duration
+ * between the initial and final speeds is equal to the parameter duration. This is done
+ * assuming constant acceleration over the given displacement.
```

Comment:
We're assuming constant acceleration over the displacement. Do we not need to give the acceleration as a parameter?

Reply:
Not in this case. The acceleration can only be pulled *lower* by this function so it should operate within a safe level without knowing that the acceleration is specifically